### PR TITLE
Return HTTP error 401 for authorization errors

### PR DESCRIPTION
**Pull request by hotzenklotz**
Until now, we redirected an unauthorised request to the dashboard / landing page which makes no sense in the context of an request API. Instead the server should return an HTTP 401 error and the client can handle that accordingly.

### Mailable description of changes:
- Prominently display an error notification when a user is logged out / not authorized to view a resource.

### Steps to test:
- Open wK in two tabs
- Log out in the first tab
- Try to navigate the some admin page in the seconds tab. While trying to download the required information for users / projects / etc an error notification should pop up. The view should not finish loading. (Permanent loading spinner)

### Questions:
1. Does this break anything? 
2. Is there a way in the client to handle 401 and act accordingly? The `Request` lib could handle all 401 errors, but that would mean a very tight integration with `redux` to logout the existing user and the react router for redirecting to the login view. Can we catch the error somewhere else instead? The router?

------
- [x] Ready for review

### CODE REVIEW

File: app/utils/ErrorHandler.scala [hunk 1]
```diff
@@ -27,6 +27,6 @@ class ErrorHandler @Inject() (
     with SecuredErrorHandler with I18nSupport {
 
   override def onNotAuthenticated(request: RequestHeader, messages: Messages): Option[Future[Result]] = {
-    Some(Future.successful(Redirect(Authentication.getLoginRoute())))
+    Some(Future.successful(Unauthorized(Messages("user.notAuthorised"))))
```
Comment (normanrz) [inline app/utils/ErrorHandler.scala]: I believe `Authentication.getLoginRoute` isn't used anywhere else and should be removed.

File: conf/messages [hunk 2]
```diff
@@ -76,7 +76,7 @@ user.configuration.dataset.updated=Dataset configuration updated
 user.experience.set=Set experience of {0}
 user.experience.removed=Removed experience from {0}
 user.email.invalid=At least one Email is invalid
-user.notAuthorised=Missing Authorisation
+user.notAuthorised=You are not authorized to view this information
```
Comment (normanrz) [inline conf/messages]: Maybe this message could be a bit more helpful: `You are currently not authorized. Please login.`

**Comment by daniel-wer** [comment]
I tested the branch and didn't notice any errors. Only weird thing is, as @hotzenklotz said before, that the user is not logged out and redirected to the login page on a 401 error.
I tried to accomplish this by throwing the 401 as an error, upgrading to react v16 and using an error boundary in the router, but I think `Uncaught (in promise)` errors somehow don't trigger the `componentDidCatch` hook, at least I can't get it to work. If someone knows better or wants to have a look, I committed the WIP to the `401-react-v16` branch.

**Comment by normanrz** [comment]
I wouldn't redirect the user to login automatically. When the user is logged out of the system for whatever reason, they might want to (manually) save any data before being navigated away.

**Comment by daniel-wer** [comment]
What kind of data are you thinking of? Any requests to download a tracing or so will fail.
The user should at least be displayed as logged out right? Currently, it's not even possible to click on login as the user is still displayed as logged in UI-wise.

**Comment by rschwanhold** [comment]
The class ""ErrorHandler" in package "utils" does override the "onNotAutenticated"-method from silhouette. We can also override "onNotAuthorised" if that is what you are looking for.

**Comment by normanrz** [comment]
@daniel-wer When we built client-side nml serializer, you could download the nml.
Maybe we could put a link to login in the toast message

**Comment by hotzenklotz** [comment]
No approvals?

**Pull request merged**